aten/src/ATen/core/op_registration/op_registration.h

```diff
@@ -66,7 +67,47 @@ class C10_API RegisterOperators final {
     return std::move(*this);
   }
 
-  // TODO Add deprecated function and lambda based kernel APIs
+  // TODO allow input schema to be just the operator name + overload name, in that case use schema generated from kernel function
+
+  /**
+   * Deprecated. For backwards compatibility only.
```

Comment:
I'd be less scary in description. For most of user defined ops extra lambda is nothing in the grand scheme of things. Maybe say that's it's not the most efficient and thus criticals ops shouldn't go through it but it's a convenient api for simple user op registration?

Reply:
I also want to push non-critical ops towards the new API. Not because they need the performance, but because it's better to have a clear answer when people ask themselves "which API should I use?". The new API looks a little worse because of the decltype, but aesthetics aren't a strong argument.